src/Store/SingleTableStore.php

```diff
@@ -211,4 +213,26 @@ private static function shortName(string $name): string
 
         return $shortName;
     }
+
+    /**
+     * @param array<string, mixed> $result
+     *
+     * @return array<string, mixed>
+     */
+    public static function normalizeResult(AbstractPlatform $platform, array $result): array
+    {
+        if (!$result['recordedOn']) {
+            return $result;
+        }
+
+        $recordedOn = Type::getType(Types::DATETIMETZ_IMMUTABLE)->convertToPHPValue(
+            $result['recordedOn'],
+            $platform
+        );
+
+        assert($recordedOn instanceof DateTimeImmutable);
```

Comment:
i really dislike `assert`. An if checks that throws or `webmozart/assert` is in my opinion more suitable.